src/classes/HH_HouseholdNaming_BATCH.cls

```diff
@@ -85,7 +85,7 @@ public class HH_HouseholdNaming_BATCH implements Database.Batchable<SObject> {
     * @return void  
     ********************************************************************************************************/
     public void execute(Database.BatchableContext bc, Sobject[] result) {
-        HH_ProcessControl.inFutureContext = true;     
+        HH_ProcessControl.setRecursionFlag(true);
```

Comment:
This is basically the same as with `ALLO_MakeDefaultAllocations_BATCH`. Once the batch iteration ends the execution context ends so any static vars are cleared. Doesn't hurt to reset either, though not explicitly required.